### PR TITLE
Add logging for debugging purpose.

### PR DESCRIPTION
**Pull request by zheng-da**
*Description of changes:*
Print more debug information and avoid printing information in all processes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/inference/ep_infer.py [hunk 1]
```diff
@@ -140,11 +139,11 @@ def infer(self, loader, save_embed_path, save_prediction_path=None,
...
-                pred_data = DistTensor(pred_shape, dtype=pred.dtype,
-                                       name='predict-'+'-'.join(etype),
-                                       part_policy=g.get_edge_partition_policy(etype),
-                                       # TODO: this makes the tensor persistent in memory.
-                                       persistent=True)
+                pred_data = create_dist_tensor(pred_shape, dtype=pred.dtype,
+                                               name='predict-'+'-'.join(etype),
+                                               part_policy=g.get_edge_partition_policy(etype),
+                                               # TODO: this makes the tensor persistent in memory.
+                                               persistent=True)
```
Comment (classicsong) [inline python/graphstorm/inference/ep_infer.py:146]: Can we make the default value of persistent to True as in most of the cases, it is set to True.

File: python/graphstorm/inference/np_infer.py [hunk 2]
```diff
@@ -123,13 +123,15 @@ def infer(self, loader, save_embed_path, save_prediction_path=None,
...
-                ntype_emb = DistTensor((g.num_nodes(ntype), embs[ntype].shape[1]),
-                                       dtype=embs[ntype].dtype, name=f'gen-emb-{ntype}',
-                                       part_policy=g.get_node_partition_policy(ntype),
-                                       # TODO: this makes the tensor persistent in memory.
-                                       persistent=True)
+                ntype_emb = create_dist_tensor((g.num_nodes(ntype), embs[ntype].shape[1]),
+                                               dtype=embs[ntype].dtype, name=f'gen-emb-{ntype}',
+                                               part_policy=g.get_node_partition_policy(ntype),
+                                               # TODO: this makes the tensor persistent in memory.
+                                               persistent=True)
```
Comment (classicsong) [inline python/graphstorm/inference/np_infer.py:134]: ditto

File: python/graphstorm/model/gnn_encoder_base.py [hunk 3]
```diff
@@ -142,11 +142,11 @@ def dist_minibatch_inference(g, gnn_encoder, get_input_embeds, batch_size, fanou
...
             # Create dist tensor to store the output embeddings
-            out_embs[ntype] = DistTensor((g.number_of_nodes(ntype), h_dim),
-                                         dtype=th.float32, name='h-last',
-                                         part_policy=g.get_node_partition_policy (ntype),
-                                         # TODO(zhengda) this makes the tensor persistent in memory.
-                                         persistent=True)
+            out_embs[ntype] = create_dist_tensor((g.number_of_nodes(ntype), h_dim),
+                                                 dtype=th.float32, name='h-last',
+                                                 part_policy=g.get_node_partition_policy (ntype),
+                                                 # TODO(zhengda) this makes the tensor persistent.
```
Comment (classicsong) [inline python/graphstorm/model/gnn_encoder_base.py:148]: ditto

**Review by classicsong — APPROVED** [review]

File: python/graphstorm/inference/ep_infer.py (hunk 1 above)
Comment (zheng-da) [reply python/graphstorm/inference/ep_infer.py:146]: we may want to remove this argument.

**Pull request merged**